include/GenSync/Syncs/BloomFilter.h

```diff
@@ -0,0 +1,231 @@
+        {
+            numHashes = nHash;
+            return *this;
+        }
+
+        /**
+         * Sets the number of expected elements in the BloomFilter.
+         * Meant to be used with setFalsePosProb setter. Cannot be used with other setters.
+         * Discards all elements inserted before the resize.
+         */
+        Builder& setNumExpElems(size_t numElems)
+        {
+            numExpElems = numElems;
+            return *this;
+        }
+
+        /**
+         * Sets the probability of false positives in the BloomFilter
+         * Meant to be used with setNumExpElems setter. Cannot be used with other setters.
+         * Discards all elements inserted before reset.
+         */
+        Builder& setFalsePosProb(float prob)
+        {
+            falsePosProb = prob;
+            return *this;
+        }
+        
+    private:
+        // Length of the BloomFilter in bits
+        size_t bfSize;
+
+        // Number of hash functions used by the BloomFilter
+        int numHashes;
+
+        // Number of expected elements in BloomFilter
+        size_t numExpElems;
+
+        // The probability/rate of false positives
+        float falsePosProb;
+    };
+
+    // default constructor
+    BloomFilter();
+
+    // default destructor
+    ~BloomFilter();
+
+    /**
+     * Getter for size of BloomFilter, specifically its length in bits.
+     * @return size_t The length of the BloomFilter's bit string
+     */
+    size_t getSize();
+
+    /**
+     * Getter for number of hash functions used by BloomFilter.
+     * @return int The number of hash functions used by BloomFilter per element
+     */
+    int getNumHashes();
+
+    /**
+     * Getter for BloomFilter's bits in form of vector<bool>.
+     * @return vector<bool> The bit string
+     */
+    vector<bool> getBits();
+
+    /**
+     * Calculates BloomFilter's rate of false positives.
+     * @param numExpElems The expected number of elements stored in the BloomFilter
+     * @return float The false positive probability/rate
+     */
+    float getFalsePosProb(size_t numExpElems);
+
+    /**
+     * Insert an element into BloomFilter.
+     * @param value The element to be added to BloomFilter
+     */
+    void insert(ZZ value);
+    
+    /**
+     * Insert a set of elements into BloomFilter.
+     * @param tarSet target set to be added to BloomFilter
+     */
+    void insert(multiset<shared_ptr<DataObject>> tarSet);
+
+    /**
+     * Probabilistically determine whether an element is in BloomFilter's bit string.
+     * @param value The element to be determined if in BloomFilter
+     * @return true iff element is probably in BloomFilter, false if definitely not.
+     */
+    bool exist(ZZ value);
+
+    /**
+     * Convert BloomFilter to a readable string.
+     * @return string
+     */
+    string toString() const;
+
+    /**
+     * Convert BloomFilter to a ZZ.
+     * @return ZZ
+     */
+    ZZ toZZ();
+
+    /**
+     * Convert ZZ to BloomFilter.

```

Comment:
You should specify something about the nature of this conversion ... I would say that this is simply the inverse operation of toZZ(), or the like.